File: Frontend/web-crawler/src/app/models/web-site-record.ts
import { Moment } from 'moment';
import { WebSiteExecutionStatus } from './execution-status';

export enum WebSiteEditMode {
  EDIT = 'edit',
  NEW = 'new',
}

export interface WebSiteRecord {
  id?: number;

  label: string;
  periodicityMinutes: number;
  url: string;
  boundaryRegExp: string;
  tags: string[];
  isActive: boolean;

  lastExecution?: Moment | null;
  executionStatus?: WebSiteExecutionStatus;
}

File: WebCrawler/Entities/WebSiteRecord.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace WebCrawler.Entities
{
    public class WebSiteRecord
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(255)]
        public required string Url { get; set; }

        [Required]
        [MaxLength(255)]
        public required string BoundaryRegExp { get; set; }

        [Required]
        public int PeriodicityMinutes { get; set; }

        [MaxLength(255)]
        public required string Label { get; set; }

        public bool IsActive { get; set; }

        public List<Tag> Tags { get; set; } = new List<Tag>();
        public List<ExecutionRecord> Executions { get; set; } = new List<ExecutionRecord>();
    }
}
